### ISSUE TITLE
Need way to add Permission-Policy directives associated with an API to its sidebar

### ISSUE THREAD

**Issue by hamishwillee** [mdn/rari#346]
### Summary

API sidebars are generated from GroupData.json, which has keys for Interfaces, external methods, properties, events, etc. This is supposed to allow all associated guide and reference for an API to be displayed in the sidebar.

However an API can be gated by Permission-Policy directives too, and there is no way to include that information in the sidebar for an API. 

The specific case I am interested in is adding [screen-wake-lock](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy/screen-wake-lock) to [Screen_Wake_Lock_API](https://developer.mozilla.org/en-US/docs/Web/API/Screen_Wake_Lock_API), but there are many cases.

### URL

https://developer.mozilla.org/en-US/docs/Web/API/Screen_Wake_Lock_API

### Reproduction steps

NA

### Expected behavior

I should be able to add a permission policy directive into the GroupData.json and have it appear in the sidebar.

### Actual behavior

I can't

### Device

Desktop

### Browser

Firefox

### Browser version

Stable

### Operating system

Windows

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/orgs/mdn/discussions/categories/platform).

**Comment by Rumyra** [comment]
@mdn/engineering could we add a 'related' key to groupdata.json which adds a section to the sidebar?

**Comment by LeoMcA** [comment]
> could we add a 'related' key to groupdata.json which adds a section to the sidebar?

@mdn/content-team Yes, we could, though there's a small potential for confusion between a "Related" section, and the "Related pages for X" section which appears in some sidebars, e.g. https://developer.mozilla.org/en-US/docs/Web/API/WakeLockSentinel - unless you wanted to fold those links in with that section?

We could group by page types under the `related` key, and map paths which are the `http-permissions-policy-directive` type into a specific "Permissions-Policy Directives" section - I think I'd prefer that to adding a dedicated `permissions-policy-directive` key for this (and the next thing that needs some grouped entries in a sidebar, and the thing after that).

This would also open up the possibility of removing the separate `guides:`/`tutorial:` keys in the future, merging them into `related` and just matching on page type.

**Comment by hamishwillee** [comment]
> This would also open up the possibility of removing the separate `guides:`/`tutorial:` keys in the future, merging them into `related` and just matching on page type.

@LeoMcA I'm not sure how that would work if I just want to add one (or two) Permissons-Policy directives to a particular API sidebar? For that I need a specific name not a page ype right?

@

**Comment by LeoMcA** [comment]
@hamishwillee the page type would just be something to group by, you'd put the slug under the `related:` key. rari would do something like:

look-up each slug under `related` -> map each page type to a sidebar section -> group by sidebar section -> create sidebar

**Comment by Rumyra** [comment]
> there's a small potential for confusion between a "Related" section, and the "Related pages for X" section which appears in some sidebars, e.g. [developer.mozilla.org/en-US/docs/Web/API/WakeLockSentinel](https://developer.mozilla.org/en-US/docs/Web/API/WakeLockSentinel) - unless you wanted to fold those links in with that section?

Yes this is what I meant - those extras in groupdata would then be shown under the related pages section - I also like the idea of grouping by page type
